### DISCUSSION TITLE
Problem mit dem Adapter und der EMMA

### DISCUSSION THREAD

**Discussion by Speedy0123** [bolliy/ioBroker.sun2000#126] [Q&A] [answered]
Moin moin.

Wir haben seid gestern eine Huawei Anlage mit EMMA am laufen.
Ich war in den Einstellungen der EMMA und habe dort MODBUS für TCP freigegeben und die IP von meinem Server eingetragen.

Im Adapter (0.15.1) Habe ich EMMA ausgewählt und bei Geräte IP eigentlich die von der EMMA eingetragen.
Im Unifi Adapter wird mir nur ein Gerät von Huawei (HEMS) angezeigt, von daher denke ich das es das richtige ist.

Wenn ich den Adapter starte bekomme ich folgende Fehlermeldung:

`
sun2000.0
	2024-10-30 14:56:50.235	info	Open Connection...
sun2000.0
	2024-10-30 14:56:50.235	warn	Error while reading from 192.168.1.190 [Reg: 30000, Len: 71, modbusID: 1] with: connect ECONNREFUSED 192.168.1.190:502
sun2000.0
	2024-10-30 14:56:50.235	error	Can not connect to Huawei inverter for modbus ID 1!
sun2000.0
	2024-10-30 14:56:50.234	warn	Adjustment: It has difficulty calibrating. The current step is 0
sun2000.0
	2024-10-30 14:56:50.233	warn	Couldnt connect Modbus TCP to 192.168.1.190:502 connect ECONNREFUSED 192.168.1.190:502
`

**Comment by bolliy** [comment, marked as answer]
@Speedy0123 bitte überprüfe die 
Voraussetzungen:
• Installer access/support für EMMA
• Die neuste EMMA firmware (V100R024C00SPC101 oder höher)
• ioBroker sollte im selben (V)LAN sein wie EMMA
• Aktivierung von Modbus auf der EMMA: Via the Fusion Solar app with the installer under Services > "Commissioning the device ” in the Settings > Communication configuration > Enable Modbus TCP connection, **TLS off und setze die ioBroker ip Adresse als client IP address**.

LG Stephan

**Reply by Speedy0123** [reply]
Ich bin der Meinung das ich das alles habe, nur das meine EMMA via WLAN im Netzwerk ist. Aber im selben wie auch der Server mit ioBrocker

**Reply by bolliy** [reply]
Ist TLS Off und die client IP auf ioBroker eingestellt?
Kannst du die EMMA vom ioBroker anpingen?
Welche FIRMWARE Version hat die EMMA? Per WLAN muss die V100R024C00SPC101 eingespielt sein!

Die EMMA scheint keine Verbindung per modbus tcp anzunehmen.

**Reply by Speedy0123** [reply]
Dann wird es daran liegen, ich habe die Firmware 100 am Ende. 
Wie bekomme ich die neue auf die EMMA?
Über Installateur auf Aktualisierung zeigt er mir keine Updatedatei an mit 101 :(

**Reply by bolliy** [reply]
https://support.huawei.com/enterprise/en/fusionsolar/smarthems-pid-256539983/software/263209812

LG Stephan

**Reply by Speedy0123** [reply]
So etwas länger gedauert das update drauf zu bekommen... 
Aber es geht immer noch nicht.
`<html><body>

sun2000.0 | 2024-10-30 19:06:20.010 | warn | Error  while reading from 192.168.1.190 [Reg: 30000, Len: 71, modbusID: 1]  with: Modbus exception 4: Slave device failure (device reports internal  error)
-- | -- | -- | --
sun2000.0 | 2024-10-30 19:06:20.010 | error | Can not connect to Huawei inverter for modbus ID 1!
sun2000.0 | 2024-10-30 19:06:20.001 | error | No device instance has been initialized!


</body>
</html>`

**Reply by bolliy** [reply]
Er findet den WR nicht. Bist du sicher, dass dein sun2000 wirklich die modbus ID = 1 hat?
Manchmal ist die ID 2 oder 16.

**Reply by Speedy0123** [reply]
kann man das irgend wo abfragen? 
Ich habe jetzt Zahlen von 1- 20 getestet aber es geht nix.

Update ist drauf habe auch noch mal geschaut ob die IP vom io stimmt TLS ist aus....

**Reply by bolliy** [reply]
https://github.com/bolliy/ioBroker.sun2000/blob/main/docs/inverter.md

**Reply by bolliy** [reply]
Hat der Adapter im Path sun2000.0.emma Werte geschrieben?

**Reply by Speedy0123** [reply]
Oh, ja waren zwischendurch nicht da.

Akku Stand und Tagesertrag... und noch  2 Werte

Auf den Wechselrichter komme ich nicht, er hat auch kein Dongle mehr dran.

**Reply by bolliy** [reply]
Sonst brauch ich mal einen debug log der ersten 30 Sekunden nach dem Start des Adapters.

**Reply by bolliy** [reply]
Dann müsste die id doch in der EMMA stehen.

**Reply by Speedy0123** [reply]
Modbus 12 ist es...

Aber Fehler:
`<html><body>

**sun2000.0 | 15756 | 2024-10-30 20:13:30.836 | error | No device instance has been initialized!**
-- | -- | -- | -- | --
sun2000.0 | 15756 | 2024-10-30 20:13:30.833 | info | Identified a Huawei SUN2000-10KTL-M1 model 429 for modbus ID 12
sun2000.0 | 15756 | 2024-10-30 20:13:30.800 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 15756 | 2024-10-30 20:13:25.753 | info | Open Connection...


</body>
</html>`

**Reply by Speedy0123** [reply]
Woran liegt das jetzt?  -_- 

`<html><body>

sun2000.0 | 15887 | 2024-10-30 20:32:10.047 | error | No device instance has been initialized!
-- | -- | -- | -- | --
sun2000.0 | 15887 | 2024-10-30 20:32:10.046 | error | No device instance has been initialized!
sun2000.0 | 15887 | 2024-10-30 20:32:00.029 | error | No device instance has been initialized!
sun2000.0 | 15887 | 2024-10-30 20:32:00.020 | error | No device instance has been initialized!
sun2000.0 | 15887 | 2024-10-30 20:31:50.043 | error | No device instance has been initialized!


</body>
</html>`

**Reply by bolliy** [reply]
Hast du mehrere IDs hinterlegt? Oder ein Komma oder Leerzeichen im Feld?

**Reply by Speedy0123** [reply]
Nein, es ist nur die 12 eingetragen bei ID nix davor und nix dahinter

**Reply by bolliy** [reply]
Sind Daten im path sun2000.0.inverter.0 ?

**Reply by Speedy0123** [reply]
Ja, Temp der Batterie und so ist da mit Werten

**Comment by bolliy** [comment]
Vielleicht hat sich in der letzten Testversion ein Bug eingenistet. Ich schau mal morgen nach...

**Reply by Speedy0123** [reply]
Okay, ich danke dir schon mal für die Geduld bis hier hin.

Werde morgen mal bei Tageslicht gucken ob die anderen Werte auch kommen. 
Einen schönen Abend noch.

**Reply by bolliy** [reply]
habe den Fehler gefunden. Kannst die 0.15.2 über npm installieren.
LG Stephan

**Reply by Speedy0123** [reply]
Jetzt kommt folgendes:

`<html><body>

sun2000.0 | 2024-10-30 22:52:00.204 | warn | Error  while reading from 192.168.1.190 [Reg: 30407, Len: 4, modbusID: 0]  with: Modbus exception 3: Illegal data value (value cannot be written to  this register)
-- | -- | -- | --



</body>
</html>`

**Reply by bolliy** [reply]
In der Version 0.15.1 habe ich auf Wunsch noch weitere Registerdaten der EMMA aufgenommen. 
Nun habe ich leider selbst keine EMMA sondern nur ein SDongle. Ich kann also die Änderungen nicht testen.
Aber vielleicht kannst du mich etwas unterstützen und einige Dinge für mich testen?

Ich habe eine neue Entwicklerversion bereitgestellt. Würdest du diese bitte einmal testen.

Installation:
Die Installation erfolgt über den Expertenmodus. Danach auf die „Krakenkatze“ klicken und dann die benutzerdefinierte Url
https://github.com/bolliy/ioBroker.sun2000/tarball/emma
eingeben und die Installation starten.
Danach muss manuell der Adapter einmal neu gestartet werden!

Nun werden die Energiemonatswerte nicht mehr gelesen. Mal sehen was passiert.

LG Stephan

**Reply by Speedy0123** [reply]
Guten Morgen, das kommt aktuell dabei raus:

`<html><body>

sun2000.0 | 2024-10-31 09:11:28.066 | info | Connected Modbus TCP to 192.168.1.190:502
-- | -- | -- | --
sun2000.0 | 2024-10-31 09:11:20.001 | info | Open Connection...
sun2000.0 | 2024-10-31 09:11:12.237 | warn | Error while reading from 192.168.1.190 [Reg: 32016, Len: 4, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:10:57.225 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 09:10:52.209 | info | Open Connection...
sun2000.0 | 2024-10-31 09:10:52.208 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:10:12.084 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 09:10:07.041 | info | Open Connection...
sun2000.0 | 2024-10-31 09:10:07.040 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:09:48.038 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 09:09:40.001 | info | Open Connection...
sun2000.0 | 2024-10-31 09:09:35.023 | warn | Error while reading from 192.168.1.190 [Reg: 37200, Len: 3, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:09:17.087 | info | Identified a Huawei SUN2000-10KTL-M1 model 429 for modbus ID 12
sun2000.0 | 2024-10-31 09:09:16.904 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 09:09:10.865 | info | Open Connection...
sun2000.0 | 2024-10-31 09:09:10.746 | info | starting.  Version 0.15.2 (non-npm: bolliy/ioBroker.sun2000#emma) in  /opt/iobroker/node_modules/iobroker.sun2000, node: v20.18.0,  js-controller: 6.0.11
sun2000.0 | 2024-10-31 09:09:05.774 | info | Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
sun2000.0 | 2024-10-31 09:09:05.773 | info | terminating
sun2000.0 | 2024-10-31 09:09:05.773 | info | cleaned everything up...
sun2000.0 | 2024-10-31 09:09:05.772 | info | Got terminate signal TERMINATE_YOURSELF
sun2000.0 | 2024-10-31 09:09:05.764 | info | New modbus settings are stored.
sun2000.0 | 2024-10-31 09:09:05.763 | info | Adjustment: It was completed successfully with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:51.681 | info | Adjustment: It has reached the step 9 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:41.017 | info | Adjustment: It has reached the step 8 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:30.014 | info | Adjustment: It has reached the step 7 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:24.990 | info | Adjustment: It has reached the step 6 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:24.352 | info | Adjustment: It has reached the step 5 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:21.782 | info | Adjustment: It has reached the step 4 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:21.585 | info | Adjustment: It has reached the step 3 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:14.185 | info | Adjustment: It has reached the step 2 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:13.540 | info | Adjustment: It has reached the step 1 with delay value 0 ms
sun2000.0 | 2024-10-31 09:08:13.462 | info | Identified a Huawei SUN2000-10KTL-M1 model 429 for modbus ID 12
sun2000.0 | 2024-10-31 09:08:11.391 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 09:08:06.306 | info | Open Connection...
sun2000.0 | 2024-10-31 09:08:06.304 | info | Adjustment: It starts for the Modbus connection...
sun2000.0 | 2024-10-31 09:08:06.199 | info | starting.  Version 0.15.2 (non-npm: bolliy/ioBroker.sun2000#emma) in  /opt/iobroker/node_modules/iobroker.sun2000, node: v20.18.0,  js-controller: 6.0.11
sun2000.0 | 2024-10-31 09:07:50.273 | info | terminating
sun2000.0 | 2024-10-31 09:07:49.771 | info | Terminated (ADAPTER_REQUESTED_TERMINATION): Without reason
sun2000.0 | 2024-10-31 09:07:49.770 | info | terminating
sun2000.0 | 2024-10-31 09:07:49.770 | info | cleaned everything up...
sun2000.0 | 2024-10-31 09:07:49.769 | info | Got terminate signal TERMINATE_YOURSELF
sun2000.0 | 2024-10-31 09:07:49.758 | error | *** Adapter deactivated, Adapter Settings incomplete! ***


</body>
</html>`

**Reply by Speedy0123** [reply]
`<html><body>

sun2000.0 | 2024-10-31 09:11:12.237 | warn | Error while reading from 192.168.1.190 [Reg: 32016, Len: 4, modbusID: 12] with: Timed out
-- | -- | -- | --
sun2000.0 | 2024-10-31 09:10:52.208 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:10:07.040 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:09:35.023 | warn | Error while reading from 192.168.1.190 [Reg: 37200, Len: 3, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:07:49.758 | error | *** Adapter deactivated, Adapter Settings incomplete! ***


</body>
</html>`

Dann scheint ruhe zu sein.

**Reply by Speedy0123** [reply]
Es kommt aber auch nichts vom Dach gerade... eine Werte bzw die Tageswerte sind rot

**Reply by bolliy** [reply]
ja, die modbus Schnittstelle der WR sind etwas sensibel. Gerade morgens wenn der WR/Speicher durch die Sonne aufwachen.

**Reply by bolliy** [reply]
Die Tageswerte im collected path stellen sich erst morgen ein. Es ist eine Differenzrechnung zum Vortagswert. Das ist ok!

**Reply by bolliy** [reply]
jetzt werde ich wieder versuchen die Energiemonatswerte einzubauen...
Hast du einen SCharger an der EMMA installiert?

**Reply by Speedy0123** [reply]
Nein, noch ist keine Ladestation dran.
Die Warungen werden mehr aus 4 sind jetzt 6 
`<html><body>

sun2000.0 | 2024-10-31 09:25:10.003 | warn | Error while reading from 192.168.1.190 [Reg: 32080, Len: 2, modbusID: 12] with: Timed out
-- | -- | -- | --
sun2000.0 | 2024-10-31 09:23:37.726 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:11:12.237 | warn | Error while reading from 192.168.1.190 [Reg: 32016, Len: 4, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:10:52.208 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:10:07.040 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:09:35.023 | warn | Error while reading from 192.168.1.190 [Reg: 37200, Len: 3, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 09:07:49.758 | error | *** Adapter deactivated, Adapter Settings incomplete! ***


</body>
</html>`

**Reply by bolliy** [reply]
Hast du die FIRMWARE auf dem WR aktuell?

**Reply by Speedy0123** [reply]
Laut der Fusion App schon... aber ich glaube nicht :) die EMMA musste ich ja auch selbst updaten...

Ich suche gleich mal wo ich die Updates her bekomme...

**Reply by bolliy** [reply]
sonst kannst du auch timeout unter dem Reiter "modbus timing" auf 15000 ms setzen.

**Comment by Speedy0123** [comment]
Hast du den Sun2000 10 KTL-M1 zufällig? ich steige hier nicht ganz durch

[https://support.huawei.com/enterprise/en/category/fusionsolar-pid-1600073963553?submodel=software](url)

**Reply by bolliy** [reply]
Nein, sun2000 8 KTL-M1. Für Updates gehe ich immer mit der Fusion App über das eingebaute WLAN vom WR.

**Comment by bolliy** [comment]
So jetzt kannst du wieder eine neue Entwicklerversion ziehen. Die Version des Adapters erhöht sich nicht.

Jetzt versuche mehr Daten in einem Rutsch zu lesen ....

**Reply by Speedy0123** [reply]
über NPM richtig?

**Reply by bolliy** [reply]
wieder über die url von github. Auf npm würde eine neue Version entstehen. Was wir in diesem Stadium nicht wollen ;)

**Reply by bolliy** [reply]
Installation:
Die Installation erfolgt über den Expertenmodus. Danach auf die „Krakenkatze“ klicken und dann die benutzerdefinierte Url
https://github.com/bolliy/ioBroker.sun2000/tarball/emma
eingeben und die Installation starten.
Danach muss manuell der Adapter einmal neu gestartet werden!

**Reply by Speedy0123** [reply]
Bis jetzt nix mehr... 
Daten werden gelesen...

Ich muss mal eben weg, ich schaue später noch mal rein ob noch was gekommen ist, oder du etwas neues hast.

`<html><body>

sun2000.0 | 2024-10-31 10:56:27.340 | info | Identified a Huawei SUN2000-10KTL-M1 model 429 for modbus ID 12
-- | -- | -- | --
sun2000.0 | 2024-10-31 10:56:26.406 | info | Connected Modbus TCP to 192.168.1.190:502
sun2000.0 | 2024-10-31 10:56:21.397 | info | Open Connection...
sun2000.0 | 2024-10-31 10:56:21.287 | info | starting.  Version 0.15.2 (non-npm: bolliy/ioBroker.sun2000#emma) in  /opt/iobroker/node_modules/iobroker.sun2000, node: v20.18.0,  js-controller: 6.0.11


</body>
</html>`

**Reply by Speedy0123** [reply]
Das ist bis jetzt reingekommen

`<html><body>

sun2000.0 | 2024-10-31 12:58:30.034 | warn | Error while reading from 192.168.1.190 [Reg: 37765, Len: 2, modbusID: 12] with: Timed out
-- | -- | -- | --
sun2000.0 | 2024-10-31 12:57:31.006 | warn | Error while reading from 192.168.1.190 [Reg: 37765, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 12:23:37.754 | warn | Error while reading from 192.168.1.190 [Reg: 32064, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 11:24:03.523 | warn | Error while reading from 192.168.1.190 [Reg: 32080, Len: 2, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 11:23:35.043 | warn | Error while reading from 192.168.1.190 [Reg: 31639, Len: 52, modbusID: 0] with: Timed out
sun2000.0 | 2024-10-31 11:01:38.374 | warn | Error while reading from 192.168.1.190 [Reg: 47081, Len: 18, modbusID: 12] with: Timed out
sun2000.0 | 2024-10-31 11:00:50.610 | warn | Error while reading from 192.168.1.190 [Reg: 37765, Len: 2, modbusID: 12] with: Timed out


</body>
</html>`

**Reply by bolliy** [reply]
ok, super, dann hat das Lesen weitere Register von der EMMA funktioniert. Nun kann ich weitere Datenpunkte anlegen.

Meine Instanze zeigt auch heute timeouts an - was sonst eigentlich nicht der Fall ist. Ich glaube es liegt am schlechten Wetter ;) Also die Meldungen kannst du getrost ignorieren.

LG Stephan

**Comment by bolliy** [comment]
so jetzt kannst du wieder von **github per url** eine neue Entwicklerversion installieren. Nun sollten 3 neue Datenpunkte auftauchen.
`{
					state: {id: 'emma.BackupTimeNotificationThreshold', name: 'Backup time notification threshold', type: 'number', unit: 'min', role: 'value', desc: 'reg:30406, len:1'},
					register: {reg: 30406, type: dataType.uint16}
				},
				{
					state: {id: 'emma.energyChargedThisMonth', name: 'Energy charged this month', type: 'number', unit: 'kWh', role: 'value.power.consumption', desc: 'reg:30407, len:2'},
					register: {reg: 30407, type: dataType.uint32, gain: 100}
				},
				{
					state: {id: 'emma.energyDischargedThisMonth', name: 'Energy discharged this month', type: 'number', unit: 'kWh', role: 'value.power.consumption', desc: 'reg:30409, len:2'},
					register: {reg: 30409, type: dataType.uint32, gain: 100}
				}`

Vielleicht kannst du in den nächsten Tagen für mich prüfen, ob die Werte der EMMA plausibel sind - und ob Fehler auftauchen.

LG Stephan

**Comment by bolliy** [comment]
@Speedy0123 konntest die letzte Version schon testen? Falls keine Fehler auftauchen (außer natürlich die TimeOuts von id 12), dann würde ich gerne ein neue Testversion releasen.

Stephan

**Reply by Speedy0123** [reply]
Installiert ist sie, befasst habe ich mich bis jetzt nicht weiter. 

Aber es sind auch keine Meldungen mehr aufgetaucht.
